cmd/update.go

```diff
@@ -73,6 +78,8 @@ func newUpdateCmd() *updateCmd {
 			if len(toUpdate) == 0 {
 				log.Infof("All binaries are up to date")
 				return nil
+			} else if root.opts.dryRun {
+				return fmt.Errorf("Command aborted, dry-run mode")
```

Comment:
In my opinion, the else is not necessary here. From a readability point of view, I would write this as follows:

```suggestion
			}
			
			if root.opts.dryRun {
				return fmt.Errorf("Updates found, exit (dry-run mode).")
```

Also I feel, the error message should focus more on the fact, that there are updates, but these are not applied, because of dry run mode.

Reply:
I actually prefer to read the login in the original way. 

If there `are` updates and we're in dry-run mode, then exit with error. It makes sense to me that are grouped in the same logical block since the condition always need to execute right after checking if there are any updates. 

On the contrary, if we just leave the block as it is here, then `dryRun` condition is completely disconnected to the fact that there are updates to make. 

Thoughts?

Reply:
My thoughts are as follows:

* I don't agree, that the `if len(toUpdate) == 0 {` is the same logical block as the `if root.opts.dryRun {`. The decides, if there are updates to be made, if not, we can cleanly exit. The second guards the code section, which performs changes to the system from the code section, which does not do any modifications. Imagine, that we could decide to print the number of bytes, that are needed to be downloaded in order to perform the update. We could then implement the dry mode such that we gather this information and print it and only after this, we would interrupt the update.
* In regards to the error message / log output, *Command aborted* to me sounds like it got interrupted unexpectedly. But this is not the case. If I provide the `--dry-run` flag, I actually expect the command to exit. But maybe this is only me (not an native English speaker).

That being said, it is your repo, your tool and your code. If you prefer your version, that is fine for me as well.

Reply:
> * In regards to the error message / log output, _Command aborted_ to me sounds like it got interrupted unexpectedly. But this is not the case. If I provide the `--dry-run` flag, I actually expect the command to exit. But maybe this is only me (not an native English speaker).

I think error message changes are ok. I didn't make any thoughts on these.

Reply:
> * I don't agree, that the `if len(toUpdate) == 0 {` is the same logical block as the `if root.opts.dryRun {`. The decides, if there are updates to be made, if not, we can cleanly exit. The second guards the code section, which performs changes to the system from the code section, which does not do any modifications. Imagine, that we could decide to print the number of bytes, that are needed to be downloaded in order to perform the update. We could then implement the dry mode such that we gather this information and print it and only after this, we would interrupt the update.

It's still seems like a code smell to me regardless of this example. For me the decision to make it the same block or another one can be easily decided by one simple question. Does the dry-run mode as it is designed right now has any other reason to exist except to exist where there are updates? My answer to that question is "no". Hence, it's clearer to me if those two conditions are in the same scope (block). 

In any case, since  I believe in democracy, it's nice that we're 3 maintainers and @sirlatrom can help us untie this difference of opinions :smile:

Reply:
I prefer the separate `if` clause as it more clearly reads as a separate 'point of no return'.

Reply:
Let's go with separate if's then!